ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -115,36 +115,31 @@ build_nodal_qrule(const Order order, const unsigned int dim)
     return {};
 }
 
-inline void
-assert_qrule_is_nodal(const FEType& fe_type, const QBase* const qrule, const Elem* const elem)
+inline bool
+qrule_is_nodal(const FEType& fe_type, const QBase* const qrule)
 {
     auto fe_order = fe_type.order;
-    auto elem_type = elem->type();
+    auto elem_type = qrule->get_elem_type();
     std::vector<FEFamily> permitted_fe_families{ LAGRANGE, L2_LAGRANGE, MONOMIAL };
     TBOX_ASSERT(std::find(permitted_fe_families.begin(), permitted_fe_families.end(), fe_type.family) !=
                 permitted_fe_families.end());
-    if (fe_type.family == MONOMIAL)
-    {
-        // Monomials can only be considered nodal for piecewise constants
-        TBOX_ASSERT(fe_order == CONSTANT);
-    }
-    switch (fe_order)
-    {
-    case CONSTANT:
-        TBOX_ASSERT(qrule->type() == QGAUSS); // the midpoint rule
-        break;
-    case FIRST:
-        TBOX_ASSERT(qrule->type() == QTRAP);
-        break;
-    case SECOND:
-        TBOX_ASSERT(qrule->type() == QSIMPSON);
-        TBOX_ASSERT((elem_type == EDGE3) || (elem_type == TRI6 || elem_type == QUAD9) ||
-                    (elem_type == TET10 || elem_type == HEX27));
-        break;
-    default:
-        TBOX_ERROR("FEProjector::assert_qrule_is_nodal(): unsupported element order "
-                   << Utility::enum_to_string<Order>(fe_order) << "\n");
-    }
+        if (fe_type.family == LAGRANGE || fe_type.family == L2_LAGRANGE || fe_type.family == MONOMIAL)
```

Comment:
This returns true for all monomial types - I don't think that's what you intended.

Reply:
Ah you're right! how about now?

Reply:
I think you forgot to push an update.